### PR TITLE
Baseline-eclipse merges jdt.core config rather than replacing it

### PR DESCRIPTION
**Pull request by uschi2000**
@bdworth @cjp39 take a look please.

### CODE REVIEW

**Comment by cjp39** [comment]
Looks better, thanks! Can we remove the overwrite logic entirely?

**Comment by uschi2000** [comment]
I don't think we can remove it since the files that get written (or overwritten if they exist) don't have corresponding configuration in Gradle's eclipse plugin, AFAIK. This seems to be doing the same thing that gradle-processors does with org.eclipse.jdt.apt.core.prefs, if I understand the code right at first glance?

**Comment by cjp39** [comment]
Then thumbs up.

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineEclipse.groovy [hunk 1]
```diff
@@ -25,11 +26,61 @@ import org.gradle.plugins.ide.eclipse.EclipsePlugin
...
+    /**
+     * Copies all name/value pairs from {@code from} to {@code into}, replacing variable names (e.g., $var or ${var})
+     * occurring in values via a {@link SimpleTemplateEngine} and the given binding.
+     */
+    static def mergeProperties(Properties from, Properties into, Map binding) {
+        from.stringPropertyNames().each { property ->
+            def propertyValue = from.getProperty(property)
+            try {
+                // Not all properties are well-formed; attempt to parse and fall-back to original value.
+                // Sadly, this is outrageously slow.
```
Comment (bdworth) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineEclipse.groovy]: How slow is "outrageously slow"?

File: gradle-baseline-java-config/resources/eclipse/org.eclipse.jdt.core.prefs [hunk 2]
```diff
@@ -93,8 +92,8 @@ org.eclipse.jdt.core.formatter.comment.preserve_white_space_between_code_and_lin
 org.eclipse.jdt.core.formatter.compact_else_if=true
 org.eclipse.jdt.core.formatter.continuation_indentation=2
 org.eclipse.jdt.core.formatter.continuation_indentation_for_array_initializer=2
-org.eclipse.jdt.core.formatter.disabling_tag=@formatter\\:off
-org.eclipse.jdt.core.formatter.enabling_tag=@formatter\\:on
+# org.eclipse.jdt.core.formatter.disabling_tag=@formatter\\:off
+# org.eclipse.jdt.core.formatter.enabling_tag=@formatter\\:on
```
Comment (bdworth) [inline gradle-baseline-java-config/resources/eclipse/org.eclipse.jdt.core.prefs]: Do these changes have a different effect than deleting the lines? If not, let's delete them.

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineEclipse.groovy (hunk 1 above)
Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineEclipse.groovy]: Didn't measure, but on the order of 2-3 seconds for the 300 lines, vs "immediate" without the template scanning.

File: gradle-baseline-java-config/resources/eclipse/org.eclipse.jdt.core.prefs (hunk 2 above)
Comment (uschi2000) [inline gradle-baseline-java-config/resources/eclipse/org.eclipse.jdt.core.prefs]: Ah, my bad. Fixed.

**Pull request merged**